From c40e54c239999f9a019fc021e18b13a5083be9a5 Mon Sep 17 00:00:00 2001
From: Ovidiu Panait <ovidiu.panait@windriver.com>
Date: Thu, 6 Sep 2018 06:18:52 +0000
Subject: [PATCH] dma: xilinx: cdmatest/vdmatest: Fix softip dma test client
 compilation

Fix the following compile errors when CONFIG_XILINX_VDMATEST=y/CONFIG_XILINX_CDMATEST=y:
error: implicit declaration of function 'put_task_struct'; did you mean 'get_task_cred'? [-Werror=implicit-function-declaration]
| put_task_struct(thread->task);
| ^~~~~~~~~~~~~~~
...
error: implicit declaration of function 'get_task_struct'; did you mean 'get_task_cred'? [-Werror=implicit-function-declaration]
| get_task_struct(thread->task);
| ^~~~~~~~~~~~~~~

This is an upstream issue that was fixed as part of the following
linux-xlnx merge-commit:

commit 818f168696f561c127f161379eb5b8d1835218a2
Merge: 9c2e29b2c81d bebc6082da0a
Author: Michal Simek <michal.simek@xilinx.com>
Date:   Thu Jan 11 13:58:55 2018 +0100

Merge tag 'v4.14' into master

Linux 4.14
Changes:
...
    dma:
    - dmaengine: Fix softip dma test client compilation (by Kedar)
...

Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
 drivers/dma/xilinx/cdmatest.c |    1 +
 drivers/dma/xilinx/vdmatest.c |    1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/dma/xilinx/cdmatest.c b/drivers/dma/xilinx/cdmatest.c
index dd02828..596d083 100644
--- a/drivers/dma/xilinx/cdmatest.c
+++ b/drivers/dma/xilinx/cdmatest.c
@@ -19,6 +19,7 @@
 #include <linux/of_dma.h>
 #include <linux/platform_device.h>
 #include <linux/wait.h>
+#include <linux/sched/task.h>
 #include <linux/dma/xilinx_dma.h>
 
 static unsigned int test_buf_size = 64;
diff --git a/drivers/dma/xilinx/vdmatest.c b/drivers/dma/xilinx/vdmatest.c
index 3e2ed65..8d5e8c0 100644
--- a/drivers/dma/xilinx/vdmatest.c
+++ b/drivers/dma/xilinx/vdmatest.c
@@ -26,6 +26,7 @@
 #include <linux/platform_device.h>
 #include <linux/random.h>
 #include <linux/slab.h>
+#include <linux/sched/task.h>
 #include <linux/wait.h>
 
 static unsigned int test_buf_size = 64;
-- 
1.7.9.5

